sgx/enclave/src/attestation.rs

```diff
@@ -1,9 +1,167 @@
-use tse;
+use sgx_rand::{Rng, os::SgxRng};
+use sgx_tcrypto::{self as tcrypto, SgxEccHandle};
 use sgx_types::*;
+use std::io::{self};
+use std::ptr;
+use std::vec::Vec;
+use tse;
+
+extern "C" {
+    pub fn ocall_sgx_init_quote ( ret_val : *mut sgx_status_t,
+                  ret_ti  : *mut sgx_target_info_t,
+                  ret_gid : *mut sgx_epid_group_id_t) -> sgx_status_t;
+    pub fn ocall_get_quote (ret_val            : *mut sgx_status_t,
+                p_sigrl            : *const u8,
+                sigrl_len          : u32,
+                p_report           : *const sgx_report_t,
+                quote_type         : sgx_quote_sign_type_t,
+                p_spid             : *const sgx_spid_t,
+                p_nonce            : *const sgx_quote_nonce_t,
+                p_qe_report        : *mut sgx_report_t,
+                p_quote            : *mut u8,
+                maxlen             : u32,
+                p_quote_len        : *mut u32) -> sgx_status_t;
+}
+
+fn init_quote() -> Result<(sgx_target_info_t, sgx_epid_group_id_t), sgx_status_t> {
+    let mut target_info = sgx_target_info_t::default();
+    let mut epid_group_id = sgx_epid_group_id_t::default();
+    let mut rt = sgx_status_t::SGX_ERROR_UNEXPECTED;
+
+    let result = unsafe {
+        ocall_sgx_init_quote(&mut rt as *mut sgx_status_t,
+                             &mut target_info as *mut sgx_target_info_t,
+                             &mut epid_group_id as *mut sgx_epid_group_id_t)
+    };
+
+    println!("epid_group_id = {:?}", epid_group_id);
+
+    if result != sgx_status_t::SGX_SUCCESS {
+        println!("ocall_sgx_init_quote failed {}", result);
+        return Err(result);
+    }
+
+    if rt != sgx_status_t::SGX_SUCCESS {
+        println!("ocall_sgx_init_quote returned {}", rt);
+        return Err(rt);
+    }
+
+    return Ok((target_info, epid_group_id))
+}
+
+fn init_report(public_key: &sgx_ec256_public_t) -> sgx_report_data_t {
+    let mut report_data: sgx_report_data_t = sgx_report_data_t::default();
+
+    // Fill ecc256 public key into report_data
+    let mut pub_k_gx = public_key.gx.clone();
+    pub_k_gx.reverse();
+    let mut pub_k_gy = public_key.gy.clone();
+    pub_k_gy.reverse();
```

Comment:
Why do we need to reverse?

Reply:
Not entirely sure what this part is for tbh, it's taken from the mutual RA example. It could be a byte ordering issue, but changing this code doesn't yield a different outcome with validation or tamper checking below.

Reply:
So I did some digging, I believe this is encoding the public key in the report data as a way for an enclave to seal a secret message for another enclave (hence "mutual" ra). It's not needed for our use case so I'm going to remove it.